sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -417,17 +411,25 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    */
   override def dropGlobalTempView(viewName: String): Boolean = {
     sparkSession.sessionState.catalog.getGlobalTempView(viewName).exists { viewDef =>
-      try {
-        val plan = sparkSession.sessionState.executePlan(viewDef)
-        sparkSession.sharedState.cacheManager.uncacheQuery(
-          sparkSession, plan.analyzed, cascade = false)
-      } catch {
-        case NonFatal(_) => // ignore
-      }
+      uncacheView(viewDef)
       sessionCatalog.dropGlobalTempView(viewName)
     }
   }
 
+  private def uncacheView(viewDef: LogicalPlan): Unit = {
+    try {
+      val viewText = viewDef match {
+        case v: View => v.desc.viewText
+        case _ => None
+      }
+      val plan = sparkSession.sessionState.executePlan(viewDef)
+      sparkSession.sharedState.cacheManager.uncacheQuery(
+        sparkSession, plan.analyzed, cascade = viewText.isDefined)
```

Comment:
Why cascade behavior should be depending on if there is view text?

Reply:
We should not un-cache cache entries dependent on the view, isn't? This is original behavior.

Reply:
The cascade behavior is from https://github.com/apache/spark/pull/21594

After #30567 , SQL temp view should have the same semantic of permanent view, so should do cascade uncache.

Reply:
Yeah, I see we should do cascade uncache. The modified question is, why cascade uncache depends on if it has view text? For a view without view text, we don't do cascade uncache?

Reply:
A view without view text is dataframe temp views, or legacy SQL temp views. We shouldn't change their behavior and keep the non-cascade uncache.

Reply:
Yeah those are the two scenarios there will be no view text and hence the old behavior, which means users could still get bad results if the view go stale. Since the legacy SQL temp view is turned off in default, I don't see a big concern there. For views created with dataset API, I'm not sure there is a way to generate view text.

Reply:
Hmm, so this is not for technical reason but to keep behavior unchanged, is it correct?

Reply:
If so, can we add a few comment for that reason?

Reply:
Yes we still keep the existing behavior for the above scenarios which this PR doesn't touch. Technically we could switch them to all use cascade uncache but that is a broader behavior change and should be discussed separately.

Sure I can add some comments.